overlord/state/notices.go

```diff
@@ -349,17 +356,12 @@ func (s *State) AddNotice(userID *uint32, noticeType NoticeType, key string, opt
 		s.notices[uniqueKey] = notice
 		newOrRepeated = true
 	} else {
-		repeatOk := true
-		if options.RepeatCheck != nil {
-			var newRepeatCheckData interface{}
-			repeatOk, newRepeatCheckData, err = options.RepeatCheck(notice, options)
-			if err != nil {
-				return "", err
-			}
-			notice.setRepeatCheckValue(newRepeatCheckData)
-		} else if options.RepeatCheckData != nil {
-			notice.setRepeatCheckValue(options.RepeatCheckData)
+		repeatOk, newRepeatCheckData, err := options.runRepeatCheck(notice)
+		if err != nil {
+			return "", err
 		}
+		// Update RepeatCheckData
+		notice.setRepeatCheckValue(newRepeatCheckData)
 		// Additional occurrence, update existing notice
 		notice.occurrences++
 		if repeatOk && (options.RepeatAfter == 0 || now.After(notice.lastRepeated.Add(options.RepeatAfter))) {
```

Comment:
I still think that maybe the RepeatAfter check should be done first?

Reply:
Updated, Thanks!